### PR TITLE
Avoid dead link when squashing hierarchy

### LINKED ISSUE #71: When squashing hierarchy, remove "List of collections with docs hosted here." link
As seen on https://ansible-collections.github.io/community.dns/branch/main/ at the bottom.

CC @briantist

### PR DESCRIPTION
**Pull request by felixfontein**
Fixes #71.

### CODE REVIEW

File: src/antsibull_docs/write_docs.py [hunk 1]
```diff
@@ -650,6 +651,8 @@ async def write_plugin_lists(collection_name: str,
         disabled.  This will disable breadcrumbs but save on memory usage.
     :kwarg for_official_docsite: Default False.  Set to True to use wording specific for the
         official docsite on docs.ansible.com.
+    :kwarg squash_hierarchy: If set to ``True``, no directory hierarchy will be used.
+        Undefined behavior if documentation for multiple collections are created.
```
Comment (oraNod) [inline src/antsibull_docs/write_docs.py:655]: @felixfontein Finding it a little hard to grok this line. Am I correct that `squash_hierarchy: True` has undefined behavior if you are creating docs for multiple collections?

Comment (felixfontein) [reply src/antsibull_docs/write_docs.py:655]: Yes. That's why it's impossible to let the CLI do that:
- https://github.com/ansible-community/antsibull-docs/blob/main/src/antsibull_docs/cli/antsibull_docs.py#L121
- https://github.com/ansible-community/antsibull-docs/blob/main/src/antsibull_docs/cli/antsibull_docs.py#L150

**Review by gotmax23 — APPROVED** [review]
LGTM.

**Comment by felixfontein** [comment]
@oraNod @gotmax23 thanks for reviewing this!

**Pull request merged**